### ISSUE TITLE
Test new TPM related systemd features

### ISSUE THREAD

**Issue by tomzy-0** [Dasharo/dasharo-issues#249]
**The problem you're addressing (if any)**

In systemd 252 release there are some new, TPM related features that might be worth checking. Those are

* systemd-measure
* updated systemd-cryptsetup, systemd-cryptenroll, and systemd-creds
* others, see release notes [1]

[1] https://lists.freedesktop.org/archives/systemd-devel/2022-October/048519.html

**Describe the solution you'd like**

Research what those features can give us, and test in with DTS on some Dasharo supported platforms.

**Where is the value to a user, and who might that user be?**

**Describe alternatives you've considered**

**Additional context**

**Comment by pietrushnic** [comment]
In the past we had idea of integrating safeboot, but safeboot packages are little bit outdated and there are other problems with those.

Despite we are not huge fans of systemd it is widely used and we should consider supporting those features. If major distros will use above features by default we should make sure that Dasharo comply to those.

More to that this feature set may open doors to DRTM integration, since we can seal to DRTM PCRs.

**Comment by mkopec** [comment]
`systemd-cryptenroll` actually has been present in a number of releases now, I've been using it on Arch with LUKS bound to PCRs 0 and 2 and it works quite well. I believe sealing to DRTM PCRs is already possible

**Comment by krystian-hebel** [comment]
> I guess the biggest problem is coreboot's PCR assignment, which is very different to the TCG-specified and commonly used assignments: https://ticket.coreboot.org/issues/420

https://review.coreboot.org/c/coreboot/+/68750

**Comment by pietrushnic** [comment]
No, it was not for DTS, but DTS assisted target system provisioning.
